publish.ps1

```diff
@@ -0,0 +1,46 @@
+# **************************************************************************
+#  Copyright (c) Cloud Native Foundation.
+#  SPDX-License-Identifier: Apache-2.0
+# **************************************************************************
+
+<#
+    .SYNOPSIS
+    Publish the CloudEvents.Sdk module to PSGallery
+
+    .PARAMETER NuGetApiKey
+    PowerShell Gallery API Key to be used to publish the module
+#>
+
+param(
+    [Parameter(Mandatory = $true)]
+    [string]
+    $NuGetApiKey
+)
+
+$moduleName = 'CloudEvents.Sdk'
+$OutputDir = Join-Path $PSScriptRoot $moduleName
+
+if (-not (Test-Path $OutputDir)) {
+    New-Item -ItemType Directory -Path $OutputDir | Out-Null
+}
+
+# Build and Test before publish
+$buildScript = Join-Path $PSScriptRoot 'build.ps1'
+$failedTests = & $buildScript -OutputDir $OutputDir
+if ($failedTests -eq 0) {
+
+    # Build is successful and all tests pass
+    $env:PSModulePath += [IO.Path]::PathSeparator + $OutputDir
+
+    $localModule = Get-Module $moduleName -ListAvailable
+    $psGalleryModule = Find-Module -Name $moduleName -Repository PSGallery
+
+    # Publish only if local module version is higher than the latest available on PSGallery
+    if ( $null -ne $psGalleryModule -and `
```

Comment:
Fail (exit 1) if this returns false

Reply:
Impment throw exception with message in this case which will make the step to fail.